addons/lunch/models/lunch.py

```diff
@@ -0,0 +1,296 @@
+                                         compute='_compute_get_previous_order_ids',
+                                         default=_default_get_previous_order_ids)
+
+    company_id = fields.Many2one('res.company', related='user_id.company_id')
+    currency_id = fields.Many2one('res.currency', related='company_id.currency_id', readonly=True)
+
+
+class lunch_order_line(models.Model):
+    """
+    lunch order line: one lunch order can have many order lines
+    """
+    _name = 'lunch.order.line'
+    _description = 'lunch order line'
+
+    @api.onchange('product_id')
+    def onchange_price(self):
+        self.price = self.product_id.price
+
+    @api.multi
+    def order(self):
+        """
+        The order_line is ordered to the supplier but isn't received yet
+        """
+        self.write({'state': 'ordered'})
+        return self.mapped('order_id').update_order_state()
+
+    @api.multi
+    def confirm(self):
+        """
+        confirm one or more order line, update order status and create new cashmove
+        """
+        for orderline in self:
+            if orderline.state != 'confirmed':
+                values = {
+                    'user_id': orderline.user_id.id,
+                    'amount': -orderline.price,
+                    'description': orderline.product_id.name,
+                    'order_id': orderline.id,
+                    'state': 'order',
+                    'date': orderline.date,
+                }
+                self.env['lunch.cashmove'].create(values)
+                orderline.state = 'confirmed'
+        return self.mapped('order_id').update_order_state()
+
+
+    @api.multi
+    def cancel(self):
+        """
+        cancel one or more order.line, update order status and unlink existing cashmoves
+        """
+        self.write({'state': 'cancelled'})
+        for orderline in self:
+            orderline.cashmove.unlink()
+        return self.mapped('order_id').update_order_state()
+
+
+    name = fields.Char(string='name', related='product_id.name', readonly=True)
+
+    order_id = fields.Many2one('lunch.order', 'Order', ondelete='cascade')
+
+    product_id = fields.Many2one('lunch.product', 'Product', required=True)
+
+    category_id = fields.Many2one('lunch.product.category', string='Product Category',
+                                  related='product_id.category_id', readonly=True, store=True)
+
+    date = fields.Date(string='Date', related='order_id.date', readonly=True, store=True)
+
+    supplier = fields.Many2one('res.partner', string='Supplier', related='product_id.supplier',
+                               readonly=True, store=True)
+
+    user_id = fields.Many2one('res.users', string='User', related='order_id.user_id',
+                              readonly=True, store=True)
+
+    note = fields.Text('Note')
+    price = fields.Float('Price')
+    state = fields.Selection([('new', 'New'),
+                              ('confirmed', 'Received'),
```

Comment:
No, there is a mixing with the following terms: validated, confirmed, received. They are all used to notify that a meal has been received. I will homogenize this as soon as we have decided if the wizards should be kept or not.